src/lib/misc/roughtime/roughtime.h

```diff
@@ -0,0 +1,119 @@
+/*
+* Roughtime
+* (C) 2019 Nuno Goncalves <nunojpg@gmail.com>
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_ROUGHTIME_H_
+#define BOTAN_ROUGHTIME_H_
+
+#include <array>
+#include <chrono>
+#include <vector>
+
+#include <botan/ed25519.h>
+
+namespace Botan {
+
+class RandomNumberGenerator;
+class DataSink_Stream;
+
+namespace Roughtime {
+
+constexpr unsigned request_min_size = 1024;
+
+/**
+* An Roughtime request.
+*/
+BOTAN_PUBLIC_API(2,10)
+std::array<uint8_t, request_min_size> encode_request(const std::array<uint8_t, 64>& nonce);
```

Comment:
I'd prefer this was returning `std::vector` since (AIUI) a Roughtime request *could* (at least in theory) exceed the min length.

Reply:
> https://roughtime.googlesource.com/roughtime/+/HEAD/PROTOCOL.md
> it‘s expected that a UDP request contain exactly 1024 bytes as there’s no point padding the request to more than the minimum.

I'm fine to change it you think vector will is "just in case" for API stability.

Reply:
I did not change this one, confirm if I should.